Basic-Video-Chat-ConnectionService_FCM/app/src/main/java/com/tokbox/sample/basicvideochat_connectionservice/FcmEventSender.java

```diff
@@ -0,0 +1,184 @@
+package com.tokbox.sample.basicvideochat_connectionservice;
+
+import static com.tokbox.sample.basicvideochat_connectionservice.OpenTokConfig.API_KEY;
+import static com.tokbox.sample.basicvideochat_connectionservice.OpenTokConfig.SESSION_ID;
+import static com.tokbox.sample.basicvideochat_connectionservice.OpenTokConfig.TOKEN;
+
+import android.telecom.DisconnectCause;
+import android.util.Log;
+
+import okhttp3.*;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.io.IOException;
+
+public class FcmEventSender {
+    private static final String TAG = VonageManager.class.getSimpleName();
+    private static final String SERVER_URL = "https://example.com/notify";  // Replace with your actual server URL
+    private static final String FCM_API_URL = "https://fcm.googleapis.com/v1/projects/<YOUR_PROJECT>/messages:send"; // Replace with your actual FCM project URL
+
+    private static final FcmEventSender instance = new FcmEventSender();
+
+    public static FcmEventSender getInstance() {
+        return instance;
+    }
+
+    private FcmEventSender() {}
+
+    // IMPORTANT: This network operation should NOT run on the main thread!
```

Comment:
I assume there is one or several blocking calls, that's why you create a new thread. That said, isn't there any really asynchronous alternative?

Reply:
We are now using ExecutorService for async calls